openstack/sharedfilesystems/v2/shares/requests.go

```diff
@@ -341,3 +341,41 @@ func Shrink(client *gophercloud.ServiceClient, id string, opts ShrinkOptsBuilder
 
 	return
 }
+
+// UpdateOptsBuilder allows extensions to add additional parameters to the
+// Update request.
+type UpdateOptsBuilder interface {
+	ToShareUpdateMap() (map[string]interface{}, error)
+}
+
+// UpdateOpts contain options for updating an existing Share. This object is passed
+// to the share.Update function. For more information about the parameters, see
+// the Share object.
+type UpdateOpts struct {
+	// Share name
+	Name *string `json:"display_name,omitempty"`
```

Comment:
`CreateOpts` has both a `Name` and `DisplayName`. Are these actually aliases to the same field?

Either way, let's change this to `DisplayName` so it matches the JSON tag.

Reply:
No, they are not aliases. `create` and `get` functions do support aliases: [create](https://github.com/openstack/manila/blob/4f22d0b19e782800b06deba74e6ec15708e5db33/manila/api/v1/shares.py#L246..L255), [get](https://github.com/openstack/manila/blob/4f22d0b19e782800b06deba74e6ec15708e5db33/manila/api/v1/shares.py#L149..L156).

Let's keep common `Name` and `Description` fields names at least in gophercloud?I don't want to copy this confusion from openstack.

Reply:
The concern I have is that the field name is different than the JSON tag:

```go
Name *string `json:"display_name,omitempty"`
```

I'd much prefer changing this to:

```go
DisplayName *string `json:"display_name,omitempty"`
```

I totally agree that Manila is handling `Name` and `DisplayName` in a weird and confusing way, but I'd prefer not to have Gophercloud fix this behavior. Gophercloud is a thin layer between an app and OpenStack - warts and all. If Gophercloud would fix weird behaviors in OpenStack, there would be a few thousand more lines of code throughout it :smile: 

The better solution would be to fix this in the app - for example, Terraform. In Terraform, you can either one of `Name` or `DisplayName` and hide the confusion within the app. If this becomes repetitive between different applications, then a central, shared utility function is the next best solution.

Does this make sense?

Reply:
Ok, I'll keep display prefix.